File: fixtures/python/credential-read/positive.py
import os
import sys

def collect():
    # must fire: static credential path
    with open("~/.aws/credentials") as f:
        creds = f.read()

    # must fire as unresolved: computed target
    target = os.environ["CFG"] + "/.netrc"
    with open(target) as f:
        extra = f.read()

    return creds, extra


def collect_indirect(cfg):
    # must fire as unresolved: subscript target. Silently dropped until the
    # dynamic alternation was made consistent across the rule family — the
    # rule matched the call and then reported nothing at all, which is the
    # one outcome invariant 3 forbids.
    with open(sys.argv[1]) as f:
        from_argv = f.read()

    # must fire as unresolved: attribute target, silent for the same reason.
    with open(cfg.credentials_path) as f:
        from_attr = f.read()

    return from_argv, from_attr

File: fixtures/javascript/credential-read/positive.js
const fs = require("fs");

function collect() {
  // must fire: static credential path
  const creds = fs.readFileSync("~/.aws/credentials", "utf8");

  // must fire as unresolved: computed target
  const target = process.env.CFG + "/.netrc";
  const extra = fs.readFileSync(target, "utf8");

  return [creds, extra];
}

module.exports = { collect };

// must fire as unresolved: computed target, reached through a destructuring
// import. The member-expression form above was covered and this one was not,
// so the same read was reported or dropped depending on import style.
const { readFileSync } = require('fs');
function loadConfig(configPath) {
  return readFileSync(configPath, 'utf-8');
}
module.exports = { loadConfig };
